ir/src/tests/ir/inlining.rs

```diff
@@ -0,0 +1,125 @@
+#[cfg(test)]
+mod tests {
+    use crate::graph::pretty;
+    use crate::passes::Inlining;
+    use crate::ConstantValue;
+    use crate::MirGraph;
+    use crate::MirType;
+    use crate::MirValue;
+    use crate::Node;
+    use crate::NodeIndex;
+    use crate::Operation;
+    use crate::SpannedMirValue;
+    use air_pass::Pass;
+
+    #[test]
+    fn test_inlining() {
```

Comment:
We should test a slightly more complex module, containing multiple calls in the same block, and calls to functions which themselves contain calls, preferably with at least one of the non-leaf functions containing a comprehension that calls a function in its body. This will catch most of the edge cases that come up when inlining non-trivial programs. It is particularly important to test the interaction of comprehensions containing function calls.

In particular, my question about soundness of how calls are inlined about probably be answered by whether or not such a test produces the expected expansion of the input program.

Reply:
Will add more complex tests to assert soundness of this pass.